src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -53,20 +53,14 @@ protected override void ProcessRule(PosRuleContext context, ChainedHeader chaine
         {
             this.CheckCoinstakeIsNotNull(header);
 
-            if (header.Coinstake.IsCoinBase)
+            // In case we are in PoW era there might be no coinstake tx.
+            // We have no way of telling if the block was supposed to be PoW or PoS so attacker can trick us into thinking that all of them are PoW so no PH is required.
+            // Therefore we get no benefit from checking proven headers while we are in PoW era.
+            if (chainedHeader.Height < this.Parent.Network.Consensus.LastPOWBlock)
```

Comment:
We can still validate PH in this case, and a POW header needs to meet hash protocol requirements.